tfhe-zk-pok/src/proofs/pke.rs

```diff
@@ -507,16 +507,17 @@ pub fn prove<G: Curve>(
 
     let effective_t_for_decomposition = t >> msbs_zero_padding_bit_count;
 
+    let effective_q = decode_q(q);
```

Comment:
In it's not really the type that I want to express but the fact that in the input q, 0 actually means u64::MAX, whereas in "effective_q", 0 means 0 and u64::MAX means u64::MAX.
Shadowing the definition might be a bit dangerous because when looking at code you might skip the part where it's redefined.
But I'm ok with that if you think that's clearer.

Reply:
is q used in its non encoded form ?

Reply:
or use q and cast in the hash

Reply:
casting for the hash would result in a 0 so essentially the encoded q

for the hash it's better to have the 64 bits value I think as we don't do u128 proofs, but yeah maybe decoded_q then